### DISCUSSION TITLE
The outputs of sliding window inference aren't discrete values?

### DISCUSSION THREAD

**Discussion by rafiibnsultan** [Project-MONAI/MONAI#5904] [Q&A] [answered]
Hello,
I am very new to the field and help would be really appreciated.
I was following the swin_unetr_btcv_segmentation_3d.ipynb on a similar dataset of 3 organs. I noticed that the outputs of the sliding window inference aren't discrete values in the form so I can use one-hot encoding to convert them to (0,1,2,3) form. Instead, it has floating values of many different values and all the non-zero values are converted to 1 and I end up with values of (0,1). If I am right, shouldn't the output image would just have pixel values in a form of (0,1,2,3) indicating the labels?

**Comment by wyli** [comment, marked as answer]
yes, `post_pred = AsDiscrete(argmax=False, threshold=0.5)` as a postprocessing will do a voxel-wise thresholding at 0.5 to make them binary predictions.  `post_pred = AsDiscrete(argmax=True)` will return the integer class labels as you described.

**Reply by rafiibnsultan** [reply]
Can you explain what the argmax parameter do here?

**Reply by wyli** [reply]
Yes, it'll add 'torch.argmax' function to the pipeline, https://pytorch.org/docs/stable/generated/torch.argmax.html So, at each voxel location, it converts a vector of prediction scores into the 'most probable' class index.
